Sources/CoreCommands/Options.swift

```diff
@@ -598,28 +599,30 @@ public struct TestLibraryOptions: ParsableArguments {
           help: .private)
     public var explicitlyEnableExperimentalSwiftTestingLibrarySupport: Bool?
 
-    private func isEnabled(_ library: TestingLibrary, `default`: Bool) -> Bool {
+    private func isEnabled(_ library: TestingLibrary, `default`: Bool, swiftCommandState: SwiftCommandState) -> Bool {
         switch library {
         case .xctest:
-            explicitlyEnableXCTestSupport ?? `default`
+            if let explicitlyEnableXCTestSupport {
+                return explicitlyEnableXCTestSupport
+            }
+            if let toolchain = try? swiftCommandState.getHostToolchain(),
+               toolchain.swiftSDK.xctestSupport == .supported {
+                return `default`
+            }
+            return false
         case .swiftTesting:
-            explicitlyEnableSwiftTestingLibrarySupport ?? explicitlyEnableExperimentalSwiftTestingLibrarySupport ?? `default`
+            return explicitlyEnableSwiftTestingLibrarySupport ?? explicitlyEnableExperimentalSwiftTestingLibrarySupport ?? `default`
         }
     }
 
     /// Test whether or not a given library is enabled.
-    public func isEnabled(_ library: TestingLibrary) -> Bool {
-        isEnabled(library, default: true)
+    public func isEnabled(_ library: TestingLibrary, swiftCommandState: SwiftCommandState) -> Bool {
```

Comment:
Similar.

Why does this function exist? Is there an ABI need for it? It seems better to just use a default value for `default`.

Reply:
I'm not sure what you're asking. It exists because, tautologically, we need it to exist so we can call it.

Are you asking why we have two separate calls that both call into the private function? Explicit intent: `isEnabled()` just asks "is the library enabled?" while `isExplicitlyEnabled()` asks "did the user explicitly pass `--enable-xctest`/`--enable-swift-testing`?"

Reply:
This is an overload of `isEnabled`, which only exists to provide a value for `default`. I'm asking why have the overload rather than just use a default value for the function that this overloads.

Reply:
I don't intend for the private overload to be part of this type's interface. The private function is just the common implementation of the two public(er) interfaces. I could just as easily call it `checkEnabledButUseThisDefaultValue()`. Or `commonImplForTheIsEnabledFunctions()` or something equally silly.